apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/store.js

```diff
@@ -90,12 +89,14 @@ const actions = {
 			response,
 		};
 	},
-	*fetchSiteHasNeverPublishedPost() {
-		const response = yield apiFetchControls( { path: '/wpcom/v2/site-has-never-published-post' } );
+	*fetchShouldShowFirstPostPublishedModal() {
+		const response = yield apiFetchControls( {
+			path: '/wpcom/v2/block-editor/should-show-first-post-published-modal',
+		} );
 
 		return {
-			type: 'WPCOM_SITE_HAS_NEVER_PUBLISHED_POST_FETCH_SUCCESS',
-			value: !! response,
+			type: 'WPCOM_SET_SHOULD_SHOW_FIRST_POST_PUBLISHED_MODAL',
+			value: response.show_first_post_published_modal,
```

Comment:
It looks like this should be 
```suggestion
			value: response.should_show_first_post_published_modal,
```

Reply:
Oops, I got mixed up between `show_welcome_guide` and `should_show_first_post_published_modal`